docs/source/build.md

```diff
@@ -1,318 +1,184 @@
-```bash
-./build.sh libraft -n
-```
+The recommended way to build and install cuVS from source is to use the `build.sh` script in the root of the repository. This script can build both the C++ and Python artifacts and provides CMake options for building and installing the headers, tests, benchmarks, and the pre-compiled shared library.
 
-Once installed, `libraft` headers (and dependencies which were downloaded and installed using `rapids-cmake`) can be uninstalled also using `build.sh`:
-```bash
-./build.sh libraft --uninstall
-```
 
-### C++ Shared Library (optional)
+### C and C++ libraries
 
-A shared library can be built for speeding up compile times. The shared library also contains a runtime API that allows you to invoke RAFT APIs directly from C++ source files (without `nvcc`). The shared library can also significantly improve re-compile times both while developing RAFT and using its APIs to develop applications. Pass the `--compile-lib` flag to `build.sh` to build the library:
+The C and C++ shared libraries are built together using the following arguments to `build.sh`:
 ```bash
-./build.sh libraft --compile-lib
+./build.sh libcuvs
 ```
 
-In above example the shared library is installed by default into `$INSTALL_PREFIX/lib`. To disable this, pass `-n` flag.
+In above example the `libcuvs.so` and `libcuvs_c.so` shared libraries are installed by default into `$INSTALL_PREFIX/lib`. To disable this, pass `-n` flag.
 
-Once installed, the shared library, headers (and any dependencies downloaded and installed via `rapids-cmake`) can be uninstalled using `build.sh`:
+Once installed, the shared libraries, headers (and any dependencies downloaded and installed via `rapids-cmake`) can be uninstalled using `build.sh`:
 ```bash
-./build.sh libraft --uninstall
+./build.sh libcuvs --uninstall
 ```
 
 
-### ccache and sccache
+#### Building the Googletests
 
-`ccache` and `sccache` can be used to better cache parts of the build when rebuilding frequently, such as when working on a new feature. You can also use `ccache` or `sccache` with `build.sh`:
+Compile the C and C++ Googletests using the `tests` target in `build.sh`.
 
 ```bash
-./build.sh libraft --cache-tool=ccache
+./build.sh libcuvs tests
 ```
 
-### C++ Tests
+The tests will be written to the build directory, which is `cpp/build/` by default, and they will be named `*_TEST`.
 
-Compile the tests using the `tests` target in `build.sh`.
+It can take sometime to compile all of the tests. You can build individual tests by providing a semicolon-separated list to the `--limit-tests` option in `build.sh`. Make sure to pass the `-n` flag so the tests are not installed.
 
 ```bash
-./build.sh libraft tests
+./build.sh libcuvs tests -n --limit-tests=NEIGHBORS_TEST;CLUSTER_TEST
```

Comment:
```suggestion
./build.sh libcuvs tests -n --limit-tests= NEIGHBORS_ANN_CAGRA_TEST; CAGRA_C_TEST
```

Using examples of tests we've already checked in might be a good idea to avoid initial potential confusion for a user